gpio/CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+#
+# Copyright (c)2021- Alligator Descartes <http://www.hermitretro.com>
+#
+# This file is part of WiringPi
+#
+#     WiringPi is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU General Public License as published by
+#     the Free Software Foundation, either version 3 of the License, or
+#     (at your option) any later version.
+#
+#     WiringPi is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#
+#     You should have received a copy of the GNU General Public License
+#     along with WiringPi.  If not, see <https://www.gnu.org/licenses/>.
+
+include_directories(${PROJECT_SOURCE_DIR}/wiringPi)
+include_directories(${PROJECT_SOURCE_DIR}/devLib)
+
+add_compile_options(-D_FILE_OFFSET_BITS=64 -Wno-pointer-to-int-cast -Wno-int-to-pointer-cast)
+
+file(GLOB_RECURSE SOURCES "*.c")
```

Comment:
Don't use GLOB to find source files. As recommended against in the official docs here: https://cmake.org/cmake/help/v3.7/command/file.html

> We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate. 

Basically- CMake does not detect changes in GLOB'd source so you end up having to "touch CMakeLists.txt" in order to trigger a rebuild, which is ugly.